eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/Utils.java

```diff
@@ -89,18 +114,33 @@ else if (groups.length == 3) {
             return HTTP_CLIENT.sendAsync(request, HttpResponse.BodyHandlers.ofInputStream())
                 .thenApply(response -> {
                     if(response.statusCode() == 200) {
-                        return Utils.parsePomFileContent(response.body());
+                        try (InputStreamReader reader = new InputStreamReader(response.body())) {
+                            return Utils.parsePomFileContent(reader);
+                        }
+                        catch (IOException ex) {
+                            logger.error("Failed to read contents for {}", dependency.toString());
+                        }
                     }
 
                     return null;
                 }).join();
     }
 
-    public static BomDependencyNoVersion toBomDependencyNoVersion(BomDependency bomDependency) {
+    static BomDependencyNoVersion toBomDependencyNoVersion(BomDependency bomDependency) {
         return new BomDependencyNoVersion(bomDependency.getGroupId(), bomDependency.getArtifactId());
     }
 
-    private static List<BomDependency> parsePomFileContent(InputStream responseStream) {
+    static List<BomDependency> parsePomFileContent(String fileName) {
+        try (FileReader reader = new FileReader(fileName)) {
+            return parsePomFileContent(reader);
+        } catch (IOException exception) {
+            logger.error("Failed to read the contents of the pom file: {}", fileName);
```

Comment:
Same here - we should throw an exception and not continue with generating the BOM.

Reply:
Same as before.